sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceRDD.scala

```diff
@@ -86,6 +92,9 @@ private class PartitionIterator[T](reader: PartitionReader[T]) extends Iterator[
     if (!hasNext) {
       throw QueryExecutionErrors.endOfStreamError()
     }
+    reader.currentMetricsValues.foreach { metric =>
+      customMetrics(metric.name()).set(metric.value())
```

Comment:
Do we need to check whether `customMetrics` contains `metric.name()`?

Reply:
We can throw a user-friendly message if it doesn't.